client/src/app.jsx

```diff
@@ -62,18 +66,71 @@ function initWithAuthToken(locale) {
 function initNotAuthenticated(msg) {
   document.querySelector('#root').innerHTML = msg;
 }
+const locales = new Set(['en', 'es', 'fr']);
+function userLocale(lo) {
+  if (lo) {
+    const l = lo.toLowerCase().substring(0, 2);
+    if (locales.has(l)) {
+      return l;
+    }
+  }
+  return 'en';
+}
 
 function dispatchOnMode() {
   const queryParams = queryString.parse(location.search);
   const accessToken = queryParams.access_token;
-  if (accessToken == null) {
-    auth
-      .init()
-      .then(({ profile, env }) => initAuthenticated(profile, env))
-      .catch(err => initNotAuthenticated(err.message));
+
+  if (url.parse(location.href).pathname !== '/auth0_callback' && accessToken == null) {
+    const authz = queryString.parse(location.search).auth || 'keycloak';
+    get('/env', { auth: authz })
+    .then(
+      ({
+        body,
+      // eslint-disable-next-line consistent-return
+      }) => {
+        auth.init(body, auth.initService(body))
+        .then(({ profile, env }) => initAuthenticated(profile, env))
+        .catch(err => initNotAuthenticated(err.message));
+      });
+  } else if (accessToken != null) {
+    auth.initExport(accessToken).then(initAuthenticated(queryParams.locale));
   } else {
-    auth.initExport(accessToken).then(initWithAuthToken(queryParams.locale));
+    get('/env', { auth: 'auth0' })
+    .then(
+      ({
+        body,
+      // eslint-disable-next-line consistent-return
+      }) => {
+        const auth0 = auth.initService(body);
+        auth.setAuth0(auth0);
+        // eslint-disable-next-line consistent-return
+        auth0.parseHash({ hash: window.location.hash }, (err, authResult) => {
+          if (err) {
+            throw err;
+          }
+
+          // eslint-disable-next-line consistent-return
+          auth0.client.userInfo(authResult.accessToken, (err2, user) => {
+            if (err2) {
+              throw err2;
+            }
+            // Now you have the user's information
+            const userr = user;
```

Comment:
Why we define `userr` as a constant if we are changing it in the next line? Note the next lines change the properties of `user`, so I don't really understand why we need `userr` at all.